src/main/java/teammates/common/datatransfer/FeedbackContributionQuestionDetails.java

```diff
@@ -800,14 +800,15 @@ private boolean hasPerceivedContribution(String email, FeedbackQuestionAttribute
      */
     @Override
     public String getNoResponseTextInHtml(String giverEmail, String recipientEmail, FeedbackSessionResultsBundle bundle, FeedbackQuestionAttributes question) {
-        String noResponseHtml = "<i>" + Const.INSTRUCTOR_FEEDBACK_RESULTS_MISSING_RESPONSE + "</i>";
+        StringBuilder noResponseHtml = new StringBuilder();
+        noResponseHtml.append("<i>").append(Const.INSTRUCTOR_FEEDBACK_RESULTS_MISSING_RESPONSE).append("</i>");
         
         // in the row for the student's self response,
         // show the perceived contribution if the student has one
         if (giverEmail.equals(recipientEmail) && hasPerceivedContribution(recipientEmail, question, bundle)) {
-            noResponseHtml += getPerceivedContributionHtml(question, recipientEmail, bundle);
+            noResponseHtml.append(getPerceivedContributionHtml(question, recipientEmail, bundle));
         } 
-        return noResponseHtml;
+        return noResponseHtml.toString();
```

Comment:
Here also, I think it can be shortened like this:
```java
return "<i>" + Const.* + "<i>" + (condition ? getPerceivedContributionHtml(...) : "");
```
Not sure if it's more readable that way, though.